bridge/src/config.rs

```diff
@@ -69,6 +69,7 @@ pub struct Node {
 	pub account: Address,
 	#[cfg(feature = "deploy")]
 	pub contract: ContractConfig,
+	pub contract_address: Option<Address>,
```

Comment:
Should this be an `Option<Address>` or could we just use an `Address`?

In the situation where we build with the `deploy` feature, it is required to be `Some` (`deploy.rs` panics if either the home or foreign contract address is `None`). In the case where we don't build with any features, we still require home and foreign contract addresses to be known to create web3 log filters for the home and foreign chains in `deposity_relay.rs`, `withdraw_relay.rs`, and `withdraw_confirm.rs`.

Reply:
Good question. One thing I considered was adding a `#[cfg(not(feature = "deploy))` tag to the field and removing the `Option`. Doing so would require more tags everywhere the field was accessed though.

My understanding is (and please correct me if I'm wrong) that there isn't necessarily a 1:1 correspondence between `Config` and `Database`. Databases sometimes have their configurations loaded from stored strings, etc. I didn't see a clean solution for cases where a database configuration is loaded from an outside source because you'd then have to update the app.configuration with those values. For some time at least before the database config was parsed in, the app.config would need to have a `None` value. Alternatively, it would be possible to decouple the parts of the configuration specific to databases from app.configuration. It didn't seem like a great solution. I'm certainly open to any suggestions, there could easily be a good way to organize it that haven't thought of.